x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineLimits.java

```diff
@@ -63,8 +62,10 @@ public LogicalPlan rule(Limit limit) {
                 }
             }
         } else if (limit.child() instanceof Join join) {
-            if (join.config().type() == JoinTypes.LEFT && join.right() instanceof LocalRelation) {
```

Comment:
Wondering why the previous restriction to LocalRelation (even if that's what was expected before). Anyways, should be good this way now.

Reply:
That came from when we only had `LOOKUP` (aka :chicken: ) and `INLINESTATS` to consider. Both of these ended up with a `LocalRelation` on the right, and we didn't want to implement this in all generality because the semantics were not 100% clear.